src/components/CostCalculator.js

```diff
@@ -0,0 +1,346 @@
+    fontFamily: 'Arial, sans-serif',
+  },
+  title: {
+    fontSize: '24px',
+    fontWeight: 'bold',
+    marginBottom: '20px',
+    textAlign: 'center',
+  },
+  grid: {
+    display: 'flex',
+    flexDirection: 'column',
+    gap: '20px',
+  },
+  inputSection: {
+    backgroundColor: '#f0f0f0',
+    padding: '20px',
+    borderRadius: '8px',
+  },
+  inputTitle: {
+    fontSize: '18px',
+    fontWeight: 'bold',
+    marginBottom: '15px',
+  },
+  inputGroup: {
+    marginBottom: '15px',
+  },
+  label: {
+    display: 'block',
+    marginBottom: '5px',
+    fontWeight: 'bold',
+  },
+  input: {
+    width: '100%',
+    padding: '8px',
+    borderRadius: '4px',
+    border: '1px solid #ccc',
+  },
+  select: {
+    width: '100%',
+    padding: '8px',
+    borderRadius: '4px',
+    border: '1px solid #ccc',
+  },
+  resultsSection: {
+    flex: '1',
+  },
+  resultItem: {
+    backgroundColor: '#f9f9f9',
+    padding: '15px',
+    marginBottom: '15px',
+    borderRadius: '8px',
+  },
+  resultLabel: {
+    fontWeight: 'bold',
+    marginBottom: '5px',
+  },
+  resultValue: {
+    fontSize: '18px',
+    color: '#0066cc',
+    marginBottom: '5px',
+  },
+  resultExplanation: {
+    fontSize: '14px',
+    color: '#666',
+  },
+  assumptions: {
+    backgroundColor: '#f0f0f0',
+    padding: '15px',
+    borderRadius: '8px',
+    marginTop: '20px',
+  },
+  assumptionsTitle: {
+    fontSize: '16px',
+    fontWeight: 'bold',
+    marginBottom: '10px',
+  },
+  assumptionsList: {
+    listStyleType: 'disc',
+    paddingLeft: '20px',
+    fontSize: '14px',
+    color: '#333',
+  },
+  errorMessage: {
+    color: 'red',
+    marginBottom: '10px',
+  },
+};
+
+const CostCalculator = () => {
+  const [gasPrice, setGasPrice] = useState(1.419);
+  const [ethPrice, setEthPrice] = useState(2700);
+  const [daThroughputLog, setDAThroughputLog] = useState(Math.log10(30000));
+  const [rollupFramework, setRollupFramework] = useState("OP Stack");
+  const [results, setResults] = useState({
+    gasNeeded: 0,
+    batchPostingPeriod: 0,
+    gasTotal: 0,
+    gasTotalUsd: 0,
+    batchesPerDay: 0,
+    gasTotalDailyUsd: 0,
+    eigenDACostUSD: 0,
+    eigenDADailyCostUSD: 0,
+    totalCostPerBatchUSD: 0,
+    totalDailyCostUSD: 0,
+  });
+  const [isLoading, setIsLoading] = useState(true);
+  const [error, setError] = useState(null);
+
+  const fetchEthPrice = async () => {
+    try {
+      const response = await fetch('https://api.coingecko.com/api/v3/simple/price?ids=ethereum&vs_currencies=usd');
+      if (!response.ok) throw new Error('Failed to fetch ETH price');
+      const data = await response.json();
+      return data.ethereum.usd;
+    } catch (error) {
+      console.error('Error fetching ETH price:', error);
+      return null;
+    }
+  };
+
+  const fetchGasPrice = async () => {
+    try {
+      const response = await fetch('https://api.etherscan.io/api?module=gastracker&action=gasoracle&apikey=G9264S2QNFC197AKJC1DFAC4DBPZHHT6MM');
```

Comment:
API key is acknowledged. This is a **free API key** which has 5req/sec. The convenience of the fetching the data and having it available to the user was communicated to be important. The application will still work with default value if the API limit is reached by actual usage or exhausting. If the issue persists, the plan is to move the call to a vercel lambda code with rate limits in the front.